tests/wallet/simple_sync/test_simple_sync_protocol.py

```diff
@@ -452,3 +458,72 @@ async def test_subscribe_for_coin_id_reorg(self, wallet_node_simulator):
+        wallet_node, server_2 = wallets[0]
+        fn_server = full_node_api.full_node.server
+        wsm: WalletStateManager = wallet_node.wallet_state_manager
+
+        await server_2.start_client(PeerInfo(self_hostname, uint16(fn_server._port)), None)
+        incoming_queue, peer_id = await add_dummy_connection(fn_server, 12312, NodeType.WALLET)
+
+        wt: WalletTool = bt.get_pool_wallet_tool()
+        ph = wt.get_new_puzzlehash()
+        for i in range(0, num_blocks):
+            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+
+        await asyncio.sleep(6)
+        coins = await full_node_api.full_node.coin_store.get_coin_records_by_puzzle_hashes(False, [ph])
+        coin_spent = coins[0].coin
+        puzzle_hash = 32 * b"\0"
+        amount = 1
+        amount_bin = int_to_bytes(1)
+        hint = 32 * b"\5"
+
+        fake_wallet_peer = fn_server.all_connections[peer_id]
+        msg = wallet_protocol.RegisterForPhUpdates([hint], 0)
+        msg_response = await full_node_api.register_interest_in_puzzle_hash(msg, fake_wallet_peer)
+        assert msg_response.type == ProtocolMessageTypes.respond_to_ph_update.value
+        data_response: RespondToPhUpdates = RespondToCoinUpdates.from_bytes(msg_response.data)
+        assert len(data_response.coin_states) == 0
+
+        condition_dict = {
+            ConditionOpcode.CREATE_COIN: [
+                ConditionWithArgs(ConditionOpcode.CREATE_COIN, [puzzle_hash, amount_bin, hint])
+            ]
+        }
+        tx: SpendBundle = wt.generate_signed_transaction(
+            10,
+            wt.get_new_puzzlehash(),
+            coin_spent,
+            condition_dic=condition_dict,
+        )
+        await full_node_api.respond_transaction(RespondTransaction(tx), fake_wallet_peer)
+
+        await time_out_assert(15, tx_in_pool, True, full_node_api.full_node.mempool_manager, tx.name())
+
+        for i in range(0, num_blocks):
+            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(puzzle_hash))
+
+        all_messages = await self.get_all_messages_in_queue(incoming_queue)
+
+        notified_state = None
+
+        for message in all_messages:
+            if message.type == ProtocolMessageTypes.coin_state_update.value:
+                data_response: CoinStateUpdate = CoinStateUpdate.from_bytes(message.data)
+                notified_state = data_response
+                break
+
+        assert notified_state is not None
+        assert notified_state.items[0].coin == Coin(coin_spent.name(), puzzle_hash, amount)
+
+        msg = wallet_protocol.RegisterForPhUpdates([hint], 0)
+        msg_response = await full_node_api.register_interest_in_puzzle_hash(msg, fake_wallet_peer)
+        assert msg_response.type == ProtocolMessageTypes.respond_to_ph_update.value
+        data_response: RespondToPhUpdates = RespondToCoinUpdates.from_bytes(msg_response.data)
+        assert len(data_response.coin_states) == 1
```

Comment:
it seems like you should check the actual value of the `coin_states` too, not just that you got *something*

Reply:
added